engine/src/exec-interface-object.cpp

```diff
@@ -4704,3 +4707,126 @@ void MCObject::GetRevAvailableVariables(MCExecContext& ctxt, MCNameRef p_key, MC
     else
         r_variables = nil;
 }
+
+void MCObject::GetRevScriptDescription(MCExecContext& ctxt, MCValueRef& r_status)
+{
+    if (!getstack() -> iskeyed())
+    {
+        ctxt . LegacyThrow(EE_STACK_NOKEY);
+        return;
+    }
+    
+    bool t_success = true;
+    
+    MCAutoValueRefBase<MCScriptObjectRef> t_object_ref;
+    
+    if (t_success)
+        t_success = MCEngineScriptObjectCreate(this, 0, &t_object_ref);
+    
+    if (t_success)
+        r_status = MCEngineExecDescribeScriptOfScriptObject(*t_object_ref);
+    
+    if (t_success &&
+        !MCExtensionConvertToScriptType(ctxt, r_status))
+    {
+        t_success = false;
+    }
+}
+
+void MCObject::GetEffectiveRevScriptDescription(MCExecContext& ctxt, MCValueRef& r_descriptions)
+{
+    bool t_success;
+    t_success = true;
+    
+    MCAutoArrayRef t_descriptions;
+    t_success = MCArrayCreateMutable(&t_descriptions);
+    
+    MCObjectList *t_object_list;
+    t_object_list = nil;
+    
+    t_success = MCObjectListAppendObjectList(t_object_list, MCfrontscripts);
+    
+    for (MCObject *t_object = this; t_success && t_object != NULL; t_object = t_object -> parent)
+    {
+        t_success = MCObjectListAppendObjectAndBehaviors(t_object_list, t_object);
```

Comment:
@livecodeali now that I look at this again (was copied from revAvailableHandlers) I think two things:

1. I should make a utility function to get the object list and use it in both.
2. `t_success` here will be clobbered by the next iteration so it really should break if false each time